core/src/main/scala/cilib/Boundary.scala

```diff
@@ -0,0 +1,90 @@
+package cilib
+
+import scalaz._
+import Scalaz._
+import spire.math.Interval
+
+object Boundary {
+
+  final case class Enforce[F[_], A](f: (A, Interval[Double]) => F[A])
+  final case class EnforceTo[F[_], A](f: (A, Interval[Double], A) => F[A])
+
+  def enforce[F[_]: Applicative, A: spire.math.Numeric](x: Position[A],
+                                                        f: Enforce[F, A]): F[NonEmptyList[A]] =
+    x.pos.zip(x.boundary).traverse { case (a, b) => f.f(a, b) }
+
+  def enforceTo[F[_], A: spire.math.Numeric, B](x: Position[A], z: Position[A], f: EnforceTo[F, A])(
+      implicit F: Applicative[F]): F[NonEmptyList[A]] =
+    x.pos.zip(x.boundary).zip(z.pos).traverse {
+      case ((a, b), c) => f.f(a, b, c)
+    }
+
+  def clamp[A](implicit N: spire.math.Numeric[A]) =
+    absorb
+
+  def absorb[A](implicit N: spire.math.Numeric[A]) =
+    Enforce((a: A, b: Interval[Double]) =>
+      Need {
+        val z = N.toDouble(a)
+
+        if (z < b.lowerValue) N.fromDouble(b.lowerValue)
+        else if (z > b.upperValue) N.fromDouble(b.upperValue)
+        else a
+    })
+
+  def random[A](implicit N: spire.math.Numeric[A]) =
+    Enforce(
+      (a: A, b: Interval[Double]) =>
+        if (b.contains(N.toDouble(a))) RVar.pure(a)
+        else Dist.uniform(b).map(N.fromDouble))
+
+  def reflect[A](implicit N: spire.math.Numeric[A]) =
+    Enforce((a: A, b: Interval[Double]) =>
+      Need {
+        val z = N.toDouble(a)
+        val range = math.abs(b.upperValue - b.lowerValue)
+
+        val result =
+          if (z < b.lowerValue) N.fromDouble(b.lowerValue + (b.lowerValue - z) % range)
+          else if (z > b.upperValue) N.fromDouble(b.upperValue - (z - b.upperValue) % range)
```

Comment:
I pictured reflection to bounce between the upper and lower bound until it's valid. To do that you'd need to know how many bounces to do to know whether to come from the lower or upper side. But since we don't know how reflection is supposed to be dealt with if it's still out of bounds I'll approve it. I'll review again if you decide to change it.

Reply:
Yip, indeed. During the testing, there was a point generated with and exponent of `E208` but the interval was far far smaller. it would have taken hours to reduce to the interval.

Reply:
True, but wouldn't it be possible to calculate how many bounces with a mod, instead of using a recursive function?

Reply:
Yes, it should be - as long as the function is decreasing to ensure a solution.